src/main/java/com/artipie/Layout.java

```diff
@@ -93,7 +96,10 @@ public Pattern pattern() {
         public Optional<Key> keyFromPath(final String path) {
             final String[] parts = splitPath(path);
             final Optional<Key> key;
-            if (parts.length >= 2) {
+            if (new CondaRqPath().test(path)) {
+                // @checkstyle MagicNumberCheck (1 line)
+                key = Optional.of(new Key.From(parts[2], parts[3]));
+            } else if (parts.length >= 2) {
```

Comment:
These numbers look cryptic here, I'd recommend either move them to constants with self-descriptive names or add inline comments to specify why number is used here.

Reply:
Extended class javadoc